drivers/modem/murata-1sc.c

```diff
@@ -0,0 +1,4499 @@
+					 {"GOLD", golden_e},
+					 {"GOLDEN", golden_e},
+					 {"ICCID", iccid_e},
+					 {"IMEI", imei_e},
+					 {"IMSI", imsi_e},
+					 {"IP", ip_e},
+					 {"IP6", ip6_e},
+					 {"MSISDN", msisdn_e},
+					 {"PSM", psm_e},
+					 {"SLEEP", sleep_e},
+					 {"SSI", ssi_e},
+					 {"STAT", connsts_e},
+					 {"SIM", sim_info_e},
+					 {"VERSION", version_e},
+					 {"VER", version_e},
+					 {"WAKE", wake_e},
+					 {}};
+
+#if defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS)
+
+#define CERTCMD_WRITE_SIZE 32 + MAX_FILENAME_LEN /* assume filename maxlen = 32 */
+#define PEM_BUFF_SIZE	   3002			 /* terminate with \" & 0 */
+
+/**
+ * following struct may not have packed memory if it has something like
+ * int follow by char then int,
+ * since array of uchar plus array of uchar will be packed
+ */
+struct cert_cmd_t {
+	uint8_t cert_cmd_write[CERTCMD_WRITE_SIZE];
+	uint8_t pem_buf[PEM_BUFF_SIZE];
+};
+
+struct mdm_sock_tls_s {
+	char host[CONFIG_MURATA_MODEM_SNI_BUFFER_SZ + 1];
+	uint8_t profile;
+	bool sni_valid;
+	bool peer_verify_disable;
+	bool client_verify;
+} murata_sock_tls_info[MDM_MAX_SOCKETS] = {{{0}, 0, false, false, false}};
+#endif /* defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS) */
+
+/**
+ * @brief Convert a series of uint8_t or byte to ascii hex value in a string
+ */
+static size_t data_to_hex_str(const void *input_buf, size_t input_len, char *output_buf,
+			      size_t output_len)
+{
+	size_t i;
+
+	for (i = 0; (i < (output_len - 1) / 2) && (i < input_len); i++) {
+		snprintk(&output_buf[(i * 2)], output_len, "%02X", ((uint8_t *)input_buf)[i]);
+	}
+
+	return i * 2;
+}
+
+/**
+ * @brief Convert ascii hex number representation to uint8_t representation
+ */
+static uint8_t nibble_to_data(char nibble)
+{
+	if (nibble >= '0' && nibble <= '9') {
+		return nibble - '0';
+	} else if (nibble >= 'A' && nibble <= 'F') {
+		return nibble - 'A' + 10;
+	} else if (nibble >= 'a' && nibble <= 'f') {
+		return nibble - 'a' + 10;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Convert uint8_t or byte to a ascii hex number representation.
+ */
+static uint8_t hex_byte_to_data(const char *hex_bytes)
+{
+	return nibble_to_data(*hex_bytes) * 0x10 + nibble_to_data(*(hex_bytes + 1));
+}
+
+/**
+ * @brief Convert uint8_t to a binary ascii (characters of either 1 or 0).
+ */
+static size_t hex_str_to_data(const char *input_buf, uint8_t *output_buf, size_t output_len)
+{
+	size_t str_len = strlen(input_buf);
+	size_t i = 0;
+
+	for (i = 0; (i < output_len) && (i * 2 < str_len); i++) {
+		output_buf[i] = hex_byte_to_data(&input_buf[i * 2]);
+	}
+	return i;
+}
+
+/**
+ * @brief Convert uint8_t to a binary ascii (string with 1 and 0).
+ */
+static char *byte_to_binary_str(uint8_t byte)
+{
+	static char buf[9] = {0};
+
+	memset(buf, 0, sizeof(buf));
+	for (int i = 0; i < 8; i++) {
+		buf[7 - i] = (byte & 1 << i) ? '1' : '0';
+	}
+
+	return buf;
+}
+
+/* Modem pins - Wake Host, Wake Modem, Reset, and Reset Done */
+static const struct gpio_dt_spec wake_host_gpio =
```

Comment:
these instance specific consts should be in a instance specific const config struct